cmd/kubectl-gke-exec-auth-plugin/provider/provider.go

```diff
@@ -0,0 +1,80 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/oauth2/google"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientauth "k8s.io/client-go/pkg/apis/clientauthentication/v1beta1"
+)
+
+var (
+	// defaultScopes:
+	// - cloud-platform is the base scope to authenticate to GCP.
+	// - userinfo.email is used to authenticate to GKE APIs with gserviceaccount
+	//   email instead of numeric uniqueID.
+	defaultScopes = []string{
+		"https://www.googleapis.com/auth/cloud-platform",
+		"https://www.googleapis.com/auth/userinfo.email"}
+)
+
+// ExecCredential return an object of type ExecCredential which
+// holds a bearer token to authenticate to GKE.
+func ExecCredential() (*clientauth.ExecCredential, error) {
+	token, err := accessToken()
+	if err != nil {
+		return nil, err
+	}
+
+	ec := &clientauth.ExecCredential{
+		TypeMeta: meta.TypeMeta{
+			Kind:       "ExecCredential",
+			APIVersion: "client.authentication.k8s.io/v1beta1",
+		},
+		Status: &clientauth.ExecCredentialStatus{
+			Token: token,
+		},
+	}
+
+	return ec, nil
+}
+
+func accessToken() (string, error) {
+	token, err := gcloudAccessToken()
+	if err == nil {
```

Comment:
invert this to `err != nil`.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow